jenkins/spark-tests.sh

```diff
@@ -90,6 +90,10 @@ jps
 
 echo "----------------------------START TEST------------------------------------"
 pushd $RAPIDS_INT_TESTS_HOME
+#only run cache tests with our serializer in nightly test for Spark version >= 3.1.1
+if [ "$(printf '%s\n' "3.1.1" "$SPARK_VER" | sort -V | head -n1)" = "3.1.1" ]; then
+  spark-submit $BASE_SPARK_SUBMIT_ARGS --conf spark.sql.cache.serializer=com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer --jars $RAPIDS_TEST_JAR ./runtests.py -v -rfExXs --std_input_path="$WORKSPACE/integration_tests/src/test/resources/" -k cache_test.py
```

Comment:
nit: continue command w/ newline ` \`. Also it would be better to move this after existing tests

not sure if we would like to give it a pytest marker for better management [pytest.ini](https://github.com/NVIDIA/spark-rapids/blob/branch-0.5/integration_tests/pytest.ini)

Reply:
the spark 3.2.0 line is using the class from com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer?  I think we should create a separate one even if just a wrapper. We do this for the shuffle for instance to make it easier for users.  it seems odd to use a class from 3.1.1 with 3.2.0 for instance. This can be a separate issue though.

Reply:
Also agree, should run this last.  @NvTimLiu @pxLi we don't have any common version parsing code do we? 

Here we need ti as a separate spark-submit call because of the way the configuration is initialized, you can't change it during runtime.

Reply:
@tgravescs currently all of our integration tests depend on https://github.com/NVIDIA/spark-rapids/blob/branch-0.5/jenkins/version-def.sh for versions that actually get ENVs from pipeline setup. Can you share in which case we need to change it during runtime?

Reply:
@tgravescs  was referring to the case where we will have to match the package name based on the spark version. I have already addressed his concern but if there is a common place for the version parsing I can reuse that. 

Please take a look at the code and let me know if you have any questions.

Reply:
I am not sure I understand it correctly, need we to convert spark version to shim version as below?

    `SPARK_VER=3.2.0-SNAPSHTOT --> SHIM_VER=320`

If yes, we can add below scripts

    `SHIM_VER=${SPARK_VER/-SNAPSHTOT/} && SHIM_VER=${SHIM_VER//./}`

    `--conf spark.sql.cache.serializer=com.nvidia.spark.rapids.shims.spark"${SHIM_VER}".ParquetCachedBatchSerializer`

Reply:
I have made the change @NvTimLiu PTAL